org-code-javabuilder/lib/src/main/java/org/code/javabuilder/UserCodeCompiler.java

```diff
@@ -89,4 +92,107 @@ private CompilationTask getCompilationTask(DiagnosticCollector<JavaFileObject> d
     // create compilation task
     return compiler.getTask(null, fileManager, diagnostics, optionList, null, files);
   }
+
+  /**
+   * For specific compiler errors we want to return custom messages. Check if the given diagnostic
+   * is due to one of those errors and return the more specific message, or null if not.
+   *
+   * @param diagnostic
+   * @return custom compiler error or null
+   */
+  private String getCustomCompilerError(Diagnostic<? extends JavaFileObject> diagnostic) {
```

Comment:
Had a thought: Would it be better for us to implement Diagnostic?

Reply:
Can you expand on that? The diagnostics are returned by the compiler

Reply:
Ah nevermind. I take that back. I was thinking it could be done with some sort of custom DiagnosticListener, but you're right, the compiler controls the diagnostic.